homeassistant/components/growatt_server/sensor_types/storage.py

```diff
@@ -74,6 +74,13 @@
         native_unit_of_measurement=UnitOfPower.WATT,
         device_class=SensorDeviceClass.POWER,
     ),
+    GrowattSensorEntityDescription(
+        key="storage_solar_production_2",
+        translation_key="storage_solar_production_2",
+        api_key="ppv2",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+    ),
```

Comment:
```suggestion
        state_class=SensorStateClass.MEASUREMENT,
    ),
```

Reply:
I think this might be needed to get the sensor to appear in the energy dashboard.

Reply:
> I think this might be needed to get the sensor to appear in the energy dashboard.

Tried, doesn't work.
From my understanding, the energy dashboard only shows lifetime sensor data. As documented [here](https://www.home-assistant.io/docs/energy/faq/#troubleshooting-missing-entities), it's possible to use only sensors with class `total` or `total_increasing`. As this PV2 value is real-time data, just like PV1 and others, passing it like a lifetime data sensor would be wrong. As the energy dashboard is now, this looks to be the intended behaviour. Other than that, adding the `measurement` state class doesn't seem to have effect elsewhere, so adding doesn't seem to be necessary.

Reply:
Hi @Cris-ET sorry, you are correct about the energy dashboard, power sensors won't be displayed, only energy. I was confusing with statistics.

I would recommend including the `state_class` as that is what is stated as needed to opt-in for long term statistics, which is probably relevant for this integration. More info here:
https://developers.home-assistant.io/docs/core/entity/sensor/#long-term-statistics.

Reply:
Dave is right; in this case, it isn't for the energy dashboard but for long-term stats.

Reply:
> Hi @Cris-ET sorry, you are correct about the energy dashboard, power sensors won't be displayed, only energy. I was confusing with statistics.

No problem @davet2001 , I still gave it a shot to be sure

> I would recommend including the state_class as that is what is stated as needed to opt-in for long term statistics, which is probably relevant for this integration. More info here:
https://developers.home-assistant.io/docs/core/entity/sensor/#long-term-statistics.

Regarding this, under the section "**Entities not representing a total amount**" it says to set `state_class` to `measurement` for devices that do not fall under the classes of `energy`, `gas` or `monetary`. This entity device class is `power` so I suppose it should be fine to include the state class. However, I see this integration does not feature a `state_class` property for any of the entities except for the ones representing a total amount, in none of the inverters, even the others aside the storage inverter. Should I add this property everywhere where needed and commit everything to this same PR even if it would end up being another kind of fix?

Reply:
The state class for power sensors should me measurement. The original suggestion from Dave is correct.

Reply:
Will take a look to all the inverter entities and fix the missing property where needed, other than adding it to the new entity I've added

Reply:
Let's do that in a separate PR, this PR adds a sensor and lets keep it just at that.

We try to keep PRs small and want them to do a single thing at the time.

../Frenck

Reply:
Understood!